src/spec/integration/categoryOrderHint.spec.coffee

```diff
@@ -55,7 +55,7 @@ newCategory = (name = 'Category name', externalId = 'externalCategoryId') ->
   externalId: externalId
 
 prepareCategoryAndProduct = (done) ->
-  jasmine.getEnv().defaultTimeoutInterval = 120000 # 2mins
+  jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000 # 2mins
```

Comment:
Interesting, was this the reason for the recurring timeouts?

Reply:
Setting jasmine.getEnv().defaultTimeoutInterval is the way you're supposed to do this, according to the Net of a Million Lies, but it didn't work.  I didn't realise this until I tried running the tests on our Jenkins server (to see if the code built there).  On Travis the code build takes three and a half minutes, and I would leave it running then come back and see what had happened; on my Macbook it's no better, but on my team's Jenkins server the Jasmine tests run from start to end in forty seconds, short enough a time for me to just watch and wait, which is how I discovered the tests that were timing out were doing so after just five seconds.

Once I'd fixed the timeouts, the other failures (apart from the intermittent ones regarding character sets) went away.  Presumably when the tests were timing out, not all of their promises were being stopped, and they were later completing and interfering with the other tests.

Reply:
Good job with finding the root cause of this! How is that possible that not all promises were stopped after the first timeout?